src/defineFunction.js

```diff
@@ -1,94 +1,127 @@
-        // then that whole function will be used as the argument to
-        // \sqrt. On the other hand, the expressions
-        //   \frac \frac 1 2 3
-        // and
-        //   \frac \sqrt 1 2
-        // will fail because \frac and \frac have equal greediness
-        // and \sqrt has a lower greediness than \frac respectively. To
-        // make these parse, we would have to change them to:
-        //   \frac {\frac 1 2} 3
-        // and
-        //   \frac {\sqrt 1} 2
-        //
-        // The default value is `1`
-        greediness?: number,
-
-        // Whether or not the function is allowed inside text mode
-        // (default false)
-        allowedInText?: boolean,
-
-        // Whether or not the function is allowed inside text mode
-        // (default true)
-        allowedInMath?: boolean,
-
-        // (optional) The number of optional arguments the function
-        // should parse. If the optional arguments aren't found,
-        // `null` will be passed to the handler in their place.
-        // (default 0)
-        numOptionalArgs?: number,
-
-        // Must be true if the function is an infix operator.
-        infix?: boolean,
-    },
+    props: FunctionPropSpec,
 
     // The handler is called to handle these functions and their arguments.
-    // It receives two arguments:
-    //   - context contains information and references provided by the parser
-    //   - args is an array of arguments obtained from TeX input
-    // The context contains the following properties:
-    //   - funcName: the text (i.e. name) of the function, including \
-    //   - parser: the parser object
-    //   - lexer: the lexer object
-    // The latter three should only be used to produce error messages.
     //
     // The function should return an object with the following keys:
     //   - type: The type of element that this is. This is then used in
     //          buildHTML/buildMathML to determine which function
     //          should be called to build this node into a DOM node
     // Any other data can be added to the object, which will be passed
     // in to the function in buildHTML/buildMathML as `group.value`.
-    handler: (context: any, args: any) => T,
+    handler: ?FunctionHandler,
 
     // This function returns an object representing the DOM structure to be
     // created when rendering the defined LaTeX function.
-    htmlBuilder: (group: T, options: Options) => any,
+    // TODO: Port buildHTML to flow and make the group and return types explicit.
+    htmlBuilder?: (group: *, options: Options) => *,
 
     // This function returns an object representing the MathML structure to be
     // created when rendering the defined LaTeX function.
-    mathmlBuilder: (group: T, options: Options) => any,
-}
+    // TODO: Port buildMathML to flow and make the group and return types explicit.
+    mathmlBuilder?: (group: *, options: Options) => *,
```

Comment:
Yep, I mentioned this in [PR 865](https://github.com/Khan/KaTeX/pull/865), though I suppose you're reviewing that part here as well. I'd said that these two types are unrelated, whereas the actual truth is that handler returns a value, where as `htmlBuilder` and `mathmlBuilder` receive a `ParseNode` (IIUC) _containing_ that value; this will flesh out better once `buildHTML` and `buildMathML` are ported and we understand its types better. We are not currently set up to specify this relationship since `ParseNode` is not parameterized by its stored value type.

Reply:
I see now.  I guess that was wishful thinking on my part.